src/core/ui.js

```diff
@@ -79,6 +82,44 @@ document.documentElement.addEventListener("click", () => {
 });
 respecUI.appendChild(menu);
 
+menu.addEventListener("keydown", e => {
+  if (e.key === "Escape" && !menu.hidden) {
+    menu.classList.add("respec-hidden");
+    menu.classList.remove("respec-visible");
+    respecPill.setAttribute("aria-expanded", String(menu.hidden));
+    respecPill.focus();
+    menu.hidden = true;
+  }
+});
+
+// Code adapted from https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element
+function trapFocus(element) {
+  const focusableEls = element.querySelectorAll(
```

Comment:
We don't need this long list at least for now, although I see you want this function to be general...

Reply:
on the one hand, this covers every possible case ... hence it's quite nice for `trapFocus(modal)` below, as it means we don't need to maintain the function or think about it. On the other hand, there are definitely some things here that are not needed right now.

Related... I wonder if we could do simplify `input` to just:
```CSS 
input:not([disabled])
```

Reply:
Can we pass this as a parameter instead?

Reply:
Do you mean passing button in this case as parameter

Reply:
I mean the selector. When we're trapping focus in some element, we generally know what focusable elements are in that particular element.

Reply:
Won't make much change here.
But passing the selector everytime will make things tough for us next time we use this function
Still if you insist , I will do this

Reply:
@saschanaz WDYT?

Reply:
I vote to keep it so that we can forget what's actually in there. This should have been much easier when CSS had jQuery's `:focusable` 😔